### ISSUE TITLE
How to run retinanet with tvm

### ISSUE THREAD

**Issue by zhiqwang** [masahi/torchscript-to-tvm#1]
Hi masahi,

Thanks for sharing these great tools, I can compile tvm and test pytorch [faster-rcnn and mask-rcnn](https://tvm.apache.org/docs/tutorials/frontend/deploy_object_detection_pytorch.html#sphx-glr-tutorials-frontend-deploy-object-detection-pytorch-py) successfully, and I found you've added tools of retinanet here, but I can't get the results as faster-rcnn does.

I'm so confused here as I understand that faster-rcnn annd retinanet share the same operators?

**Comment by masahi** [comment]
Yeah, I'd love to get retinanet working but currently we have both compilation (too slow) and execution problem (the results wrong). See https://discuss.tvm.apache.org/t/error-from-compiling-and-running-retinanet-from-torchvision/8678

It seems retinanet in torchvision 0.8 release doesn't use batched nms, so it has a large loop that do small nms in each iteration. These loops are completely unrolled during tracing (unlike MaskRCNN which has for loop after tracing and conversion to TVM) so the result TVM model is huge.

So I may revisit this problem after another release. Until then I don't have to investigate this issue further.

**Comment by zhiqwang** [comment]
Hi @masahi ,

I just want to share with you of my first overture on `tvm`.

I've tested another one-stage-detector named `yolov5`, it is produced by **ultralytics** [here](https://github.com/ultralytics/yolov5). I've refactored it under the principle of **torchvision**, in other words, I used `torchvision`'s ops to rewrite `yolov5`'s modules. And with recently updates of `tvm`, and I can compile and run `yolov5` successfully now. [Here](https://github.com/zhiqwang/yolov5-rt-stack) is my repo.

I haven't tried `retinanet`, but now I believe it could also be run with minor fixes :-)

**Comment by masahi** [comment]
Nice! Do you have a ready-to-run script? I want to try it.

I'm also interested in trying out SSD. Do you know a good PyTorch SSD model that could be a good fit for TVM? (i.e., using NMS from torchvision, can be traced correctly etc).

**Comment by zhiqwang** [comment]
Just use the following scripts to load a `yolov5s`

```python
class TraceWrapper(nn.Module):
    def __init__(self, model):
        super().__init__()
        self.model = model

    def forward(self, inp):
        out = self.model(inp)
        return dict_to_tuple(out[0])

model_func = torch.hub.load('zhiqwang/yolov5-rt-stack', 'yolov5s', pretrained=True)
model = TraceWrapper(model_func)
```

The other is same with the [tutorial](https://tvm.apache.org/docs/tutorials/frontend/deploy_object_detection_pytorch.html#sphx-glr-tutorials-frontend-deploy-object-detection-pytorch-py) supplied by `tvm` teams.

I've rewrite the `mobilenetv2_ssd` also in [here](https://github.com/zhiqwang/demonet), there are some bugs before, I'm not sure whether it can be run now. It seems that it's very helpful of recently updates of `tvm`.

**Comment by masahi** [comment]
Thanks, I was able to run yolo5 and get the same output as torch, very nice! Please do let me know if SSD is also working.

**Comment by zhiqwang** [comment]
Hi @masahi , Sure!
